### PR TITLE
Fix possible content-type header duplication

### PR DESCRIPTION
**Pull request by delaguilaluis**
Fixing issue #188.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -118,14 +118,13 @@ internals.Client.prototype.request = function (method, url, options, callback, _
 
     uri.method = method.toUpperCase();
     uri.headers = options.headers || {};
-    const hasContentLength = Object.keys(uri.headers).some((key) => {
-
-        return key.toLowerCase() === 'content-length';
-    });
+    const hasContentLength = Boolean(internals.findHeader('content-length', uri.headers));
```
Comment (geek) [inline lib/index.js]: This changes the meaning a little if `content-length` is 0.  Instead we should

```
const hasContentLength = (typeof internals.findHeader('content-length', uri.headers) === 'number');
```

Comment (delaguilaluis) [inline lib/index.js]: You are right. What do you think about doing

```javascript
const hasContentLength = internals.findHeader('content-length', uri.headers) !== undefined;
```

?

Comment (geek) [inline lib/index.js]: That should be fine, I doubt we have to deal with a null... why not number?

Comment (delaguilaluis) [inline lib/index.js]: Just to determine the cases where this should be `false` based on a value and not on its type.

At the end this variable answers the question "does it has content-length?" not "does it has a content-length which type is `number`?"

File: lib/index.js [hunk 2]
```diff
@@ -557,5 +556,13 @@ internals.tryParseBuffer = function (buffer) {
     return result;
 };
 
+internals.findHeader = function (headerName, headers) {
+
+    const foundKey = Object.keys(headers)
+        .find((key) => key.toLowerCase() === headerName);
```
Comment (delaguilaluis) [inline lib/index.js]: I will update this to compare the lower case version of both values.

**Pull request merged**